lib/msf/core/db_manager/session.rb

```diff
@@ -145,61 +144,80 @@ def report_session(opts)
       end
     end
 
-
     if opts[:session]
       session.db_record = s
+      if session.assoc_exploit.user_data_is_match?
+        # do some shit with the match
+        MetasploitDataModels::AutomaticExploitation::MatchResult.create!(
+          match: session.assoc_exploit.user_data[:match],
+          match_set: session.assoc_exploit.user_data[:match_set],
+          run: session.assoc_exploit.user_data[:run],
+          state: 'succeeded',
+        )
+      elsif session.via_exploit
+        # This is a live session, we know the host is vulnerable to something.
+        infer_vuln_from_session(session, wspace)
+      end
```

Comment:
Should there be an `else` that logs we couldn't do anything or error if there should always be a `via_exploit`?